src/widget/wtrackmenu.cpp

```diff
@@ -1595,6 +1625,161 @@ void WTrackMenu::slotClearAllMetadata() {
+    const auto trackRefs = getTrackRefs();
+    QStringList locations;
+    locations.reserve(trackRefs.size());
+    for (const auto& trackRef : trackRefs) {
+        QString location = trackRef.getLocation();
+        // TODO de-duplicate when extending this feature to Playlists
+        locations.append(location);
+    }
+
+    {
+        // Set up and run the confirmation dalog
+        QDialog* dlgDelConfirm = new QDialog(nullptr);
+        dlgDelConfirm->setModal(true); // just to be sure
+        dlgDelConfirm->setWindowTitle(tr("Delete Track Files"));
+
+        QVBoxLayout* delLayout = new QVBoxLayout;
+        QLabel* delWarning = new QLabel;
+        delWarning->setText(tr("Permanently delete these files from disk?") +
+                QString("<br><br><b>") +
+                tr("This can not be undone!") + QString("</b>"));
+        delWarning->setTextFormat(Qt::RichText);
+        delWarning->setSizePolicy(QSizePolicy(QSizePolicy::Minimum,
+                QSizePolicy::Minimum));
+
+        // NOTE(ronso0) We could also make this a table to allow showing
+        // artist and title if file names don't suffice to identify tracks.
+        QListWidget* delListWidget = new QListWidget;
+        delListWidget->setSizePolicy(QSizePolicy(QSizePolicy::Minimum,
+                QSizePolicy::MinimumExpanding));
+        delListWidget->addItems(locations);
+        delListWidget->setMinimumSize(
+                mixxx::widgethelper::adjustedListWidgetSize(*delListWidget, *this));
+
+        QDialogButtonBox* delButtons = new QDialogButtonBox();
+        QPushButton* cancelBtn = delButtons->addButton(
+                tr("Cancel"),
+                QDialogButtonBox::RejectRole);
+        QPushButton* deleteBtn = delButtons->addButton(
+                tr("Delete Files"),
+                QDialogButtonBox::AcceptRole);
+        cancelBtn->setDefault(true);
+        // This is required after customizing the buttons, otherwise neither button
+        // would close the dialog.
+        connect(cancelBtn, &QPushButton::clicked, dlgDelConfirm, &QDialog::reject);
+        connect(deleteBtn, &QPushButton::clicked, dlgDelConfirm, &QDialog::accept);
+
+        delLayout->addWidget(delListWidget);
+        delLayout->addWidget(delWarning);
+        delLayout->addWidget(delButtons);
+        dlgDelConfirm->setLayout(delLayout);
+
+        if (dlgDelConfirm->exec() == QDialog::Rejected) {
+            return;
+        }
+    }
+
+    // Set up and initiate the track batch operation
+    const auto progressLabelText =
+            tr("Removing %1 track file(s) from disk...",
+                    "",
+                    getTrackCount());
+    const auto trackOperator =
+            RemoveTrackFilesFromDiskTrackPointerOperation();
+    applyTrackPointerOperation(
+            progressLabelText,
+            &trackOperator);
+
+    // Purge deleted tracks and show deletion summary message.
+    if (trackOperator.tr_tracksToPurge.length() > 0) {
```

Comment:
Please add a comment that explains why purging of tracks has to be deferred until the batch operation has finished. Either here and /or inside `RemoveTrackFilesFromDiskTrackPointerOperation`. Probably in `RemoveTrackFilesFromDiskTrackPointerOperation` because this is were it actually matters.

Reply:
done